src/FixedDataTableCellGroup.js

```diff
@@ -148,41 +160,50 @@ class FixedDataTableCellGroupImpl extends React.Component {
   render() /*object*/ {
     var props = this.props;
     var columns = props.columns;
-    var cells = new Array(columns.length);
-    var contentWidth = sumPropWidths(columns);
-
-    var currentPosition = 0;
-    for (var i = 0, j = columns.length; i < j; i++) {
-      var columnProps = columns[i].props;
-      var cellTemplate = columns[i].template;
-
-      var recyclable = _.get(
-        this.state.isCellRecyclableByColumnId,
-        [columnProps.columnKey],
-        columnProps.allowCellsRecycling
+    let columnsToRender = props.columnsToRender || [];
+
+    if (props.isScrolling) {
+      // allow static array to grow while scrolling
+      this._staticCells.length = Math.max(
```

Comment:
We maintain the dynamic list of cells to be rendered through `this._staticCells` (functionally equivalent to `this._staticRowArray` in [FixedDataTableRow](https://github.com/schrodinger/fixed-data-table-2/blob/1010d2688c943d75a937e6d4d5d8915e36932451/src/FixedDataTableBufferedRows.js#L137)).

**Recap of the logic**:
The list of columns in the visible viewport expands or contracts during scrolling, based on different column widths. 
But this leads to a problem where if we simply render the list of cells based on these visible columns, then they're going to be constantly mounted/unmounted during scrolling.

To fix this, we store the rendered cells in `this._staticCells`, and have it grow by updating elements just for the new columns.
eg: Initially we had 10 columns in the viewport. so this._staticCells will have 10 entries.
Suppose the next render had only 6 columns in the viewport. Then we only update 6 entries in `this._staticCells`. The rest of the 4 elements will still be rendered but hidden since they're outside the viewport.
This prevents any cells from being unmounted.

Once scrolling is complete, we shrink this to exactly fit the current list of columns in the viewport.